modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidgetEnhanced.php

```diff
@@ -0,0 +1,639 @@
+<?php
+
+namespace Drupal\tide_content_collection\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextareaWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\tide_automated_listing\SearchApiIndexHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Implementation of the content collection configuration enhanced widget.
+ *
+ * @FieldWidget(
+ *   id = "content_collection_configuration_enhanced",
+ *   label = @Translation("Content Collection Configuration (Enhanced)"),
+ *   field_types = {
+ *     "content_collection_configuration"
```

Comment:
@joshua-salsadigital please check the Solution Directions thoroughly. The SDs clearly require `content_collection_configuration` widget.

Reply:
Reviewed the SD and made changes as per the SD